### PR TITLE
D3d12: Perf and compat improvements

### PR DESCRIPTION
**Pull request by vlj**
This PR avoid an extra copy of index buffers. It also makes several fixes in Braid, which is still not playable but at least doesn't crash.

### CODE REVIEW

File: rpcs3/Emu/RSX/Common/BufferUtils.cpp [hunk 1]
```diff
@@ -109,34 +109,54 @@ void write_vertex_array_data_to_buffer(void *buffer, u32 first, u32 count, size_
...
+void expandIndexedTriangleFan(char *dst, u32 address, size_t indexCount)
 {
-	IndexType *typedDst = reinterpret_cast<IndexType *>(dst);
-	const IndexType *typedSrc = reinterpret_cast<const IndexType *>(src);
 	for (unsigned i = 0; i < indexCount - 2; i++)
 	{
-		typedDst[3 * i] = typedSrc[0];
-		typedDst[3 * i + 1] = typedSrc[i + 2 - 1];
-		typedDst[3 * i + 2] = typedSrc[i + 2];
+		(IndexType&)dst[(3 * i) * sizeof(IndexType)] = VMRead<IndexType>::exec(address);
```
Comment (Nekotekina) [inline rpcs3/Emu/RSX/Common/BufferUtils.cpp]: This is a bit... too much. `vm::ps3::_ref<IndexType>(address)` must be sufficient if you didn't mean something extraordinary in the VMRead template. `vm::_ref`, if it's RSX-specific file and ps3 namespace can be correctly used.

**Comment by vlj** [comment]
Use vm::ps3::_ref now

**Comment by tambry** [comment]
@vlj Needs to be rebased

**Comment by vlj** [comment]
Rebased + fixed index copying code.
Now Braid works with d3d12.

I will likely port some fixes to gl branch later.

**Comment by vlj** [comment]
Change moved to types.h

**Comment by vlj** [comment]
Can someone push this branch ?

**Comment by vlj** [comment]
rebased

**Pull request merged**